### PR TITLE
[ERA-10175] fix:simplifying schema field select widget code to allow touch events to open/close the select menu

### PR DESCRIPTION
**Pull request by JoshuaVulcan**
### What does this PR do?
- Removes legacy code which appears to be swallowing event propagation in a way that 

### Relevant link(s)
* https://allenai.atlassian.net/browse/ERA-10175
* env still building

### Any background context you want to provide(if applicable)
https://allenai.atlassian.net/browse/ERCS-2577
This impacts a number of sites who prefer to use mobile web.

### CODE REVIEW

File: src/SchemaFields/index.js [hunk 1]
```diff
@@ -258,10 +258,6 @@ export const SelectWidget = ({
 
   const [isMenuOpen, setMenuOpen] = useState(false);
 
-  const SelectContainer = (props) => <div ref={containerRef}>
```
Comment (luixlive) [inline src/SchemaFields/index.js:261]: Do we still need the `containerRef` if this was deleted?

**Review by luixlive — APPROVED** [review]
Approved. I just left an observation.

File: src/SchemaFields/index.js (hunk 1 above)
Comment (JoshuaVulcan) [reply src/SchemaFields/index.js:261]: I'm not sure 🤔 honestly I'm trying to keep the surface area of changes as small as possible

**Review by fonzy911 — APPROVED** [review]
Feature appears to be working properly now:

https://github.com/user-attachments/assets/c37eda23-c77c-4030-a170-c17f115a5f71

**Pull request merged**